src/Network/ABCI/Internal/Wire.hs

```diff
@@ -28,22 +28,19 @@ encodeLengthPrefixC = CL.map $ foldMap encodeLengthPrefix
 
 decodeLengthPrefixC
   :: Monad m
-  => ConduitT BS.ByteString (Either String [BS.ByteString]) m ()
-decodeLengthPrefixC = awaitForever $ \bytes ->
-  case splitOffMessages bytes of
-    Left err       -> yield (Left err)
-    Right messages -> yield $ Right messages
+  => ConduitT BS.ByteString (Either Error.Error [BS.ByteString]) m ()
+decodeLengthPrefixC = awaitForever $ yield . splitOffMessages
   where
-    splitOffMessages :: BS.ByteString -> Either String [BS.ByteString]
+    splitOffMessages :: BS.ByteString -> Either Error.Error [BS.ByteString]
     splitOffMessages bs
-      | bs == mempty = pure []
+      | bs == mempty = Right []
       | otherwise = do
-          n <- runParser getVarInt bs
+          n <- first (Error.ProtoLensParseError bs) $ runParser getVarInt bs
           let lengthHeader = runBuilder $ putVarInt n
           messageBytesWithTail <- case BS.stripPrefix lengthHeader bs of
-            Nothing -> let prefixWithMsg = show (cs . BS16.encode $ lengthHeader :: String , cs . BS16.encode $ bs :: String)
```

Comment:
Why was this BS16.encode used?
I  used it in showBS too if it's not needed LMK

Reply:
It's used because if you print raw `ByteString` it comes out looking like nonsense, for readablity people always use an encoding, and to me `Hex` encoding is the best for humans